src/Utilities/regExpInsensitiveFilter.js

```diff
@@ -0,0 +1,47 @@
+// run a predicate function over each key of an object
+// const hasValue = f => o =>
+//  Object.keys(o).some(x => f(o[x]))
+const hasValue = f => o => {
+    let key;
+    for (key in o) {
+        if (f(o[key])) {return true;}
+
+        ;
+    }
+
+    return false;
+};
+
+// convert string to regular expression
+const toReg = str =>
+    new RegExp(str?.replace(/\//g, '//'), 'gi');
+
+// test a string with a regular expression
+const match = reg => x =>
+    reg?.test(x);
+
+// filter an array by a predicate
+// const filter = f => a => a.filter(a)
+const filter = f => a => {
```

Comment:
.... is this reimplementing `Array.filter`?

Reply:
Please, do not re-implement a native function. It's doubtful that it is more performant than a native implementation, even if it is more performant it's probably not worth the confusion it contributes.

Reply:
@bastilian I rewrote the function